vllm/sequence.py

```diff
@@ -5,24 +5,22 @@
 from array import array
 from collections import defaultdict
 from dataclasses import dataclass
-from typing import TYPE_CHECKING, Any, Callable, Dict, List, Mapping, Optional
+from typing import Any, Callable, Dict, List, Mapping, Optional
 from typing import Sequence as GenericSequence
 from typing import Set, Tuple, Union, cast
 
 import msgspec
 import torch
 
+from vllm.inputs import LLMInputs
 from vllm.inputs.parse import is_valid_encoder_decoder_llm_inputs
 from vllm.lora.request import LoRARequest
+from vllm.multimodal import MultiModalDataDict, MultiModalPlaceholderDict
 from vllm.pooling_params import PoolingParams
 from vllm.prompt_adapter.request import PromptAdapterRequest
 from vllm.sampling_params import SamplingParams
 from vllm.spec_decode.metrics import SpecDecodeWorkerMetrics
 
-if TYPE_CHECKING:
-    from vllm.inputs import LLMInputs
-    from vllm.multimodal.base import MultiModalDataDict
-
```

Comment:
I prefer using `TYPE_CHECKING` imports here rather than in each `multimodal` file to avoid circular imports. Is there a particular reason why you swapped them?

Reply:
Yeah this was unfortunate and I wasn't sure what the best way to proceed was. The issue is that `msgpack` uses type annotations at runtime, so the `MultiModalPlaceholderDict` type needed to be properly imported in `SequenceGroupMetadata`. The options that came to mind were:

 - Replace the proper type annotation with `Dict[Any]` in `SequenceGroupMetadata`
 - Lift the type definition out of `vllm.multimodal` to somewhere else
 - Invert the `TYPE_CHECKING` guards (what I ended up doing)

But I'm happy to do whatever.

Reply:
That makes sense. I'm fine with this then.